test/integration/controller/jobs/xgboostjob/xgboostjob_controller_test.go

```diff
@@ -0,0 +1,575 @@
+				gomega.Expect(k8sClient.Get(ctx, wlLookupKey, createdWorkload)).Should(gomega.Succeed())
+				return apimeta.FindStatusCondition(createdWorkload.Status.Conditions, kueue.WorkloadPodsReady)
+			}, util.Timeout, util.Interval).Should(gomega.BeComparableTo(podsReadyTestSpec.wantCondition, ignoreConditionTimestamps))
+		},
+		ginkgo.Entry("No progress", podsReadyTestSpec{
+			wantCondition: &metav1.Condition{
+				Type:    kueue.WorkloadPodsReady,
+				Status:  metav1.ConditionFalse,
+				Reason:  "PodsReady",
+				Message: "Not all pods are ready or succeeded",
+			},
+		}),
+		ginkgo.Entry("Running XGBoostJob", podsReadyTestSpec{
+			jobStatus: kftraining.JobStatus{
+				Conditions: []kftraining.JobCondition{
+					{
+						Type:   kftraining.JobRunning,
+						Status: corev1.ConditionTrue,
+						Reason: "Running",
+					},
+				},
+			},
+			wantCondition: &metav1.Condition{
+				Type:    kueue.WorkloadPodsReady,
+				Status:  metav1.ConditionTrue,
+				Reason:  "PodsReady",
+				Message: "All pods were ready or succeeded since the workload admission",
+			},
+		}),
+		ginkgo.Entry("Running XGBoostJob; PodsReady=False before", podsReadyTestSpec{
+			beforeCondition: &metav1.Condition{
+				Type:    kueue.WorkloadPodsReady,
+				Status:  metav1.ConditionFalse,
+				Reason:  "PodsReady",
+				Message: "Not all pods are ready or succeeded",
+			},
+			jobStatus: kftraining.JobStatus{
+				Conditions: []kftraining.JobCondition{
+					{
+						Type:   kftraining.JobRunning,
+						Status: corev1.ConditionTrue,
+						Reason: "Running",
+					},
+				},
+			},
+			wantCondition: &metav1.Condition{
+				Type:    kueue.WorkloadPodsReady,
+				Status:  metav1.ConditionTrue,
+				Reason:  "PodsReady",
+				Message: "All pods were ready or succeeded since the workload admission",
+			},
+		}),
+		ginkgo.Entry("Job suspended; PodsReady=True before", podsReadyTestSpec{
+			beforeJobStatus: &kftraining.JobStatus{
+				Conditions: []kftraining.JobCondition{
+					{
+						Type:   kftraining.JobRunning,
+						Status: corev1.ConditionTrue,
+						Reason: "Running",
+					},
+				},
+			},
+			beforeCondition: &metav1.Condition{
+				Type:    kueue.WorkloadPodsReady,
+				Status:  metav1.ConditionTrue,
+				Reason:  "PodsReady",
+				Message: "All pods were ready or succeeded since the workload admission",
+			},
+			jobStatus: kftraining.JobStatus{
+				Conditions: []kftraining.JobCondition{
+					{
+						Type:   kftraining.JobRunning,
+						Status: corev1.ConditionFalse,
+						Reason: "Suspended",
+					},
+				},
+			},
+			suspended: true,
+			wantCondition: &metav1.Condition{
+				Type:    kueue.WorkloadPodsReady,
+				Status:  metav1.ConditionFalse,
+				Reason:  "PodsReady",
+				Message: "Not all pods are ready or succeeded",
+			},
+		}),
+	)
+})
+
+var _ = ginkgo.Describe("Job controller interacting with scheduler", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+	const (
+		instanceKey = "cloud.provider.com/instance"
```

Comment:
nit: this is declared at the top already as labelKey

Reply:
Oh, you are right. Thanks!

Reply:
It seems that we have the same issue in all frameworks (including batch/job, jobset...) :(
So, I will create a separate PR to fix this issue before moving this forward.

Reply:
sgtm